src/pushprocessor.ts

```diff
@@ -174,9 +206,21 @@ export class PushProcessor {
 
         // Merge the client-level defaults with the ones from the server
         const globalOverrides = newRules.global.override;
-        for (const override of DEFAULT_OVERRIDE_RULES) {
+        for (let override of DEFAULT_OVERRIDE_RULES) {
             const existingRule = globalOverrides.find((r) => r.rule_id === override.rule_id);
 
+            // Dynamically add the user ID as the value for the is_user_mention rule.
+            if (override.rule_id === RuleId.IsUserMention) {
+                // If the user ID wasn't provided, skip the rule.
+                if (!userId) {
+                    continue;
+                }
+
+                override = JSON.parse(JSON.stringify(override)); // deep clone
```

Comment:
I think our support policy still says last 3 versions of Safari, which rules out https://developer.mozilla.org/en-US/docs/Web/API/structuredClone which was introduced in Safari 15.4. Otherwise I'd be recommending it :-(

Reply:
FTR I copied this code from a few lines above. 😄